docs/basic/binary.md

```diff
@@ -48,19 +68,25 @@ The bisection method turns a problem of finding extreme values into a decision p
 
 ### Binary Search in STL
 
-One thing worth noting is that for an ordered array you can use `std::lower_bound()` to find [the first number greater than or equal to your value](http://www.cplusplus.com/reference/algorithm/lower_bound/), and `std::upper_bound()` to find the [first greater than the number of your value](https://en.cppreference.com/w/cpp/algorithm/upper_bound).
+One thing worth noting is that for an ordered array you can use `std::lower_bound()` to find [the first number greater than or equal to your value](http://www.cplusplus.com/reference/algorithm/lower_bound/), and `std::upper_bound()` to find the [first greater than the number of your value](https://en.cppreference.com/w/cpp/algorithm/upper_bound). Both of them are defined in the header `<algorithm>`.
 
-Please note that it must be an ordered array, otherwise the answer is wrong.
+Please note that it must be an ordered array, as both of them are implemented in binary search algorithm; otherwise the answer would be unexpected. <!---修正了一点点语法问题, 以及, 感觉使用Unexpected更合适-->
 
-For detailed usage, please refer to [STL](../lang/csl/index.md) 。
+For detailed usage, please refer to [STL](../lang/csl/index.md) .
 
 ### Bisection answer
 
 When solving problems, we often consider enumerating answers and then check whether the value is correct. If we replace the enumeration here with a bisection method, it becomes a "bisection answer".
 
-Let’s take a look at a sample problem [Luogu P1873 Cut Tree](https://www.luogu.com.cn/problem/P1873) (original link in Chinese). We can enumerate the answers from 1 to 1000000000 (1 billion), but this naive approach certainly would not receive full marks. Because running from 1 to 1 billion is too time-consuming. We can divide the answer from 1 to 1 billion, and check the feasibility every time (usually using the greedy method). **This is the bisection answer**.
+Let's take a look at a sample problem [Luogu P1873 Cut Tree](https://www.luogu.com.cn/problem/P1873) (original link in Chinese). 
 
-The example solution for practice problem is listed below: 
+<!---咱觉得可以加一个题目摘要-->
+???+note "abstract of the problem"
+    You are given two integer variables $n$ ($1 \le n \le 10^6$) and $m$($1 \le m \le 2 \times 10^9$) and an array $a_1, a_2, ..., a_n (a_i \le 10^9)$. You are asked to find the minimum integer $h$ which satisfies that $$\sum a_i - h \space (\forall i \in [1, n],  a_i > h > m)$$ 
+
+We can enumerate the answers from 1 to 1000000000 (1 billion), but this naive approach certainly would not receive full marks. Because running from 1 to 1 billion is too time-consuming. We can divide the answer from 1 to 1 billion, and check the feasibility every time (usually using the greedy method). **This is the bisection answer**.
```

Comment:
```suggestion
We can enumerate the answers from $1$ to $10^9$ (1 billion), but this brute force approach certainly would not receive full marks since enumerating from $1$ to 10^9$ billion is too time-consuming. We can bisect the answer within the range of $[1,~10^9]$, and check the feasibility every time (usually using the greedy method). **This is the bisection answer algorithm**.
```